standard/expressions.md

```diff
@@ -2073,11 +2144,13 @@ Unlike the syntactically equivalent *null_conditional_member_access* or *null_co
 
 ```ANTLR
 null_conditional_invocation_expression
-    : null_conditional_member_access '(' argument_list? ')'
-    | null_conditional_element_access '(' argument_list? ')'
+    : null_conditional_member_access null_forgiving_operator? '(' argument_list? ')'
+    | null_conditional_element_access null_forgiving_operator? '(' argument_list? ')'
     ;
 ```
 
+<!-- [Placeholder] 12.8.11 -->
+<!-- [Placeholder] 12.8.11 End -->
```

Comment:
### Placeholder 12.8.11

#### First Alternative

A certain compiler produces a compiler time error in the following situation.

---
&lt;!-- [ToDo] C#9’s function pointers are also excluded, as the following restriction is stated  in terms of what is included the text will probably be fine but this will need to be confirmed
-->
The optional *null_forgiving_operator* may be included if and only if the *null_conditional_member_access* or
*null_conditional_element_access* has a *delegate_type*.

---

(As before the [ToDo] comment is part of the suggestion.)

#### Second Alternative

---
Remove placeholder

---

Reply:
Just to check we're thinking about the same thing:

```csharp
ActionHolder? x = null;
x?.Action!(); // This is valid
x?.Dynamic!(); // This is not (CS8598)

record ActionHolder(Action? Action, dynamic? Dynamic);
```

Reply:
Basically I suspect this should go hand-in-hand with "Placeholder 12.8.10.1" - if we go with the first alternative there, we should go with it here too, and vice versa.